tethys_cli/gen_commands.py

```diff
@@ -77,23 +77,94 @@ def add_gen_parser(subparsers):
+    gen_parser.add_argument('--workspaces-root', dest='workspaces_root',
+                            help='For production. Path to workspaces diretory. Defaults to ${TETHYS_HOME}/workspaces. '
+                                 'Applies default if directory does not exist.')
+    gen_parser.add_argument('--bypass-portal-home', dest='bypass_portal_home', action='store_true',
+                            help='Bypasses the Tethys home page. Defaults to False')
+    gen_parser.add_argument('--add-quota-handlers', dest='add_quota_handlers', nargs='+',
+                            help='Append one or more dot-formatted handlers to the resource quota handlers list in '
+                                 'settings.py. Defaults to tethys_quotas.handlers.workspace.WorkspaceQuotaHandler. '
+                                 'e.g.: tethysapp.dam_inventory.dam_quota_handler.DamQuotaHandler')
+    gen_parser.add_argument('--django-analytical', dest='django_analytical', nargs='+',
+                            help='Provide one or more ID:SERVICE_ID pair for django analytical options in settings.py. '
+                                 'All IDs default to False. Available IDs are: CLICKMAP_TRACKER_ID, CLICKY_SITE_ID, '
+                                 'CRAZY_EGG_ACCOUNT_NUMBER, GAUGES_SITE_ID, GOOGLE_ANALYTICS_JS_PROPERTY_ID, '
+                                 'GOSQUARED_SITE_TOKEN, HOTJAR_SITE_ID, HUBSPOT_PORTAL_ID, INTERCOM_APP_ID, '
+                                 'KISSINSIGHTS_ACCOUNT_NUMBER, KISSINSIGHTS_SITE_CODE, KISS_METRICS_API_KEY, '
+                                 'MIXPANEL_API_TOKEN, OLARK_SITE_ID, OPTIMIZELY_ACCOUNT_NUMBER, PERFORMABLE_API_KEY, '
+                                 'PIWIK_DOMAIN_PATH, PIWIK_SITE_ID, RATING_MAILRU_COUNTER_ID, SNAPENGAGE_WIDGET_ID, '
+                                 'SPRING_METRICS_TRACKING_ID, USERVOICE_WIDGET_KEY, WOOPRA_DOMAIN, '
+                                 'YANDEX_METRICA_COUNTER_ID. '
+                                 'e.g.: CLICKMAP_TRACKER_ID:123456 CLICKY_SITE_ID:789123')
+    gen_parser.add_argument('--add-backends', dest='add_backends', nargs='+',
+                            help='Add one or more authentication backends to settings.py. Django try these backends in '
+                                 'the same order they are listed. Provide the dot-formatted python path to a custom '
+                                 'backend or one of the following keys: hydroshare, linkedin, google, facebook. '
+                                 'The default backends are django.contrib.auth.backends.ModelBackend and '
+                                 'guardian.backends.ObjectPermissionBackend. '
+                                 'e.g.: project.backend.CustomBackend hydroshare')
+    gen_parser.add_argument('--oauth-options', dest='oauth_options', nargs='+',
+                            help='Add options for oauth providers in the settings.py. '
```

Comment:
Add options for oauth providers in the settings.py. -> Provide KEY:VALUE pairs of parameters for oauth providers in the settings.py.